internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -101,10 +108,37 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 		"capacity": {
 			Type:         pluginsdk.TypeInt,
 			Optional:     true,
-			Default:      20,
+			ExactlyOneOf: []string{"auto_scale_profile", "capacity"},
```

Comment:
Hi @stephybun Thanks for looking into this.

The default capacity that was added here https://github.com/hashicorp/terraform-provider-azurerm/pull/24033 has been causing some issues now that the API supports autoscaling. To be specific, if a user with the `capacity` field set, moves over to use `auto_scale_profile`, applies the terraform change, and then reruns plan, they see an unusual diff where capacity is being changed from 0 -> 20. This is undesirable behavior and an application would succeed and drop autoscaling (and the plan/apply loop continues).

- Is there a way to handle such a behavior in Terraform?
- I understand it's a breaking change, and wanted to know the policy around it as well. You are saying we can only do this on TF version 4.0.0? Given the error handling will call out that the user needs to set the `one of` fields, it should guide them to do so and their deployments would continue to work after that?

Reply:
> Is there a way to handle such a behavior in Terraform?

In cases where setting a new property on a resource changes the value of an existing property with a default, such as this one we advise users to apply `ignore_changes` on the existing property affected when configuring/setting the new property. We usually add this as a note under the property in the documentation.

> I understand it's a breaking change, and wanted to know the policy around it as well. You are saying we can only do this on TF version 4.0.0?

We can only remove the default for `capacity` and set `ExactlyOneOf` on `capacity` and the `auto_scale_profile` block in a major version, so in 4.0. What we usually do is to wrap behaviours that are breaking and should only be applied in the next major release behind the feature flag `features.FourPointOh()`, you can find examples on how to do this by searching for the flag in the code base. It can be a bit tedious to do for schema changes so if you have issues I'm happy to push that change to your branch for you, just let me know.

> Given the error handling will call out that the user needs to set the one of fields, it should guide them to do so and their deployments would continue to work after that?

There are often cases where the Azure API doesn't play nice with the way Terraform works and we have to go down the route of error'ing a users config or allowing a diff in order to guide users to make appropriate changes to fix their deployments. However I don't think that applies here, we can avoid breaking users that haven't set the `capacity` property by wrapping the removal of the default and applying `ExactlyOneOf` behind the feature flag. This will let us support the new `auto_scale_profile` and not force users to make a configuration change if they don't want to configure `auto_scale_profile`.

Reply:
Thanks a lot for these details @stephybun 

I chatted with some members of our team internally and we like the idea of going with `ignore_changes` for the time being. I have updated the PR to remove the breaking change. Can you please take a look? Thanks!!

Reply:
Going to resolve this given we are not doing a breaking change.